lib/reek/source/source_code.rb

```diff
@@ -36,14 +36,14 @@ def initialize(code, origin, parser: Parser::Ruby22)
       # @param source [File|IO|String] - the given source
       #
       # @return an instance of SourceCode
-      # :reek:DuplicateMethodCall: { max_calls: 2 }
       def self.from(source)
-        case source
-        when File     then new(source.read, source.path)
-        when IO       then new(source.readlines.join, IO_IDENTIFIER)
-        when Pathname then new(source.read, source.to_s)
-        when String   then new(source, STRING_IDENTIFIER)
-        end
+        params = case source
+                 when File     then { code: source.read,           origin: source.path }
+                 when IO       then { code: source.readlines.join, origin: IO_IDENTIFIER }
+                 when Pathname then { code: source.read,           origin: source.to_s }
+                 when String   then { code: source,                origin: STRING_IDENTIFIER }
+                 end
+        new(params)
```

Comment:
Well, imagine you'd have to do a second thing based on the switch. This is a boilerplate code spawn kitchen :D